sdk/python/kfp/compiler/compiler_utils.py

```diff
@@ -348,6 +373,11 @@ def _get_uncommon_ancestors(
     Returns:
         A tuple which are lists of uncommon ancestors for each task.
     """
+
+    if getattr(task1, 'is_exit_handler', False) or getattr(
+            task2, 'is_exit_handler', False):
+        return ([task1.name], [task2.name])
```

Comment:
As described in other comments, I'm curious if there's an approach that avoid adding special handling for the `ExitHandler` case.